datacube/model/utils.py

```diff
@@ -346,7 +350,13 @@ def visit(ds):
     if not isinstance(root, SimpleDocNav):
         root = SimpleDocNav(root)
 
-    return visit(root)
+    try:
+        return visit(root)
+    except BadMatch as e:
+        if root.id not in str(e):
+            raise BadMatch(f"Error loading lineage dataset: {e}")
```

Comment:
should this be `from None`?
```suggestion
            raise BadMatch(f"Error loading lineage dataset: {e}") from None
```

Reply:
Oh, good call.